src/simulator_core/entities/assets/pipe.py

```diff
@@ -13,78 +12,210 @@
+
+        :param EsdlAssetObject esdl_asset: The ESDL asset object associated with the
+                current pipe object.
+
+        :return: The heat transfer coefficient of the pipe [W/(m2 K)]. If the heat transfer
+                coefficient cannot be calculated - for example when the material table is
+                not specified - , the default alpha value is returned.
         """
-        pass
-
-    def get_setpoints(self, **kwargs: dict) -> Dict:
-        """Placeholder to get the setpoint attributes of an asset.
+        diameters, heat_coefficients = get_thermal_conductivity_table(esdl_asset=esdl_asset)
+        if diameters:
+            # Create a numpy array of the diameters and heat coefficients
+            diameters = np.array(diameters)
+            heat_coefficients = np.array(heat_coefficients)
+            # Calculate the heat transfer coefficient from the heat transfer table
+            heat_transfer_coefficient = np.sum(
+                calculate_heat_transfer_coefficient(
+                    inner_diameter=diameters[:-1],
+                    outer_diameter=diameters[1:],
+                    thermal_conductivity=heat_coefficients,
+                )
+            )
+            return 1 / heat_transfer_coefficient
+        else:
+            return self.alpha_value
+
+    def add_physical_data(self, esdl_asset: EsdlAssetObject) -> None:
+        """Method to add physical data to the asset.
 
-        :return Dict: The setpoints of the asset. The keys of the dictionary are the names of the
-            setpoints and the values are the values.
+        :param EsdlAssetObject esdl_asset: The ESDL asset object associated with the
+                current pipe object.
         """
-        return {}
+        # Error handling is performed in EsdlAssetObject.get_asset_parameters
+        self.length, _ = esdl_asset.get_property(
+            esdl_property_name="length", default_value=self.length
+        )
+        self.roughness, _ = esdl_asset.get_property(
+            esdl_property_name="roughness", default_value=self.length
+        )
+        self.diameter = self._get_diameter(esdl_asset=esdl_asset)
+        self.alpha_value = self._get_heat_transfer_coefficient(esdl_asset=esdl_asset)
 
     def simulation_performed(self) -> bool:
-        """Placeholder to indicate that a simulation has been performed.
+        """Check whether a simulation has been performed.
 
         :return bool: True if a simulation has been performed, False otherwise.
         """
-        return True
-
-    def create(self) -> None:
-        """Placeholder to create an asset in a pandapipes network."""
-        pass
+        if self.pandapipes_net.res_pipe[self._pipe_index] is AttributeError:
+            # TODO: Implement specific error
+            return False
+        else:
+            # Retrieve the setpoints
+            return True
 
     def write_to_output(self) -> None:
```

Comment:
for now ok, but at some point we need to decided which outputs we want to store for the pipe and also other components, maybe good to make issue out of it

Reply:
We should discuss the write_to_output functionality the dictionaries work but its not clear to me if its a good method. The dictionary structure feels really elaborate; cant we do something with dataclasses?

Reply:
[#41](https://github.com/Project-OMOTES/simulator-core/issues/41) already in an issue.